src/features/import-map.js

```diff
@@ -10,40 +10,68 @@
 import { baseUrl as pageBaseUrl, parseImportMap, resolveImportMap } from '../common.js';
 import { systemJSPrototype } from '../system-core.js';
 
-var importMap, importMapPromise;
-if (typeof document !== 'undefined') {
-  const scripts = document.getElementsByTagName('script');
-  for (let i = 0; i < scripts.length; i++) {
-    const script = scripts[i];
-    if (script.type !== 'systemjs-importmap')
-      continue;
-
-    if (!script.src) {
-      importMap = parseImportMap(JSON.parse(script.innerHTML), pageBaseUrl);
-    }
-    else {
-      importMapPromise = fetch(script.src)
-      .then(function (res) {
-        return res.json();
-      })
-      .then(function (json) {
-        importMap = parseImportMap(json, script.src);
-      });
+var acquiringImportMaps = true, importMapPromise;
```

Comment:
See https://github.com/WICG/import-maps/blob/master/spec.md#acquiring-import-maps for why I created this variable. The same thing could also be accomplished with a truthiness check on importMapPromise.

Reply:
The approach makes sense. Having `importMapPromise` be either a promise or null / undefined sounds like it would work just as well and be one less variable. Always gotta think of that footprint :P

Reply:
👍 will change